apk/apk.go

```
@@ -84,6 +98,61 @@ func (k *Apk) Icon(resConfig *androidbinary.ResTableConfig) (image.Image, error)
+
+		var innerForegroundImagePath string
+		if androidbinary.IsResID(adaptiveIcon.Background.Drawable) {
+			resID, err := androidbinary.ParseResID(adaptiveIcon.Foreground.Drawable)
+			if err != nil {
+				return nil, errorf("failed to parse resID %q: %w", adaptiveIcon.Foreground.Drawable, err)
+			}
+
+			innerForegroundImagePathTmp, err := k.table.GetResource(resID, nil)
+			switch v := innerForegroundImagePathTmp.(type) {
+			case string:
+				innerForegroundImagePath = v
```

golangci-lint

ineffectual assignment to innerForegroundImagePath

(ineffassign)